modules/ROOT/pages/patterns/shortest-paths.adoc

```diff
@@ -538,3 +538,84 @@ RETURN [stop IN nodes(p) | stop.name] AS r1,
 
 2+d|Rows: 1
 |===
+
+
+[[planning-StatefulShortestPath-operator]]
+== Planning the StatefulShortestPath-operator
+See xref:planning-and-tuning/operators/operators-detail.adoc#query-plan-stateful-shortest-path-into[the operator-detail page] for a description of the `StatefulShortestPath(All)` and `StatefulShortestPath(Into)` operators.
+
+When the subexpression of the statefulShortestPath-operator matches only one of both boundary nodes, then a `StatefulShortestPath(All)` is planned.
+
+When the subexpression of the statefulShortestPath-operator matches both boundary nodes, then a `StatefulShortestPath(Into)` is planned only when both boundary nodes match to at most one data node. Otherwise, this plan is avoided when possible.
+Another plan will be favoured where the subexpression matches only one of both boundary nodes followed by an `StatefulShortestPath(All)`.
+
+As an example, consider the case where the left boundary node matches to 10 data nodes and the right boundary node to 3 data nodes.
+Using `StatefulShortestPath(Into)` requires 30 bidirectional breath first search (BFS) invocations.
+Using `StatefulShortestPath(All)` requires only 3 unidirectional BFS invocations expanding from the right boundary node.
+
+In case you want the `StatefulShortestPath(Into)` to be chosen, then you can rewrite your query using a `CALL subquery`.
```

Comment:
```suggestion
Sometimes this heuristic decision to favour `StatefulShortestPath(All)` can lead to worse query performance.
In case you want the `StatefulShortestPath(Into)` to be chosen, then you can rewrite your query using a `CALL` subquery.
```